### PR TITLE
Feat: Added wise-lending-v2 fees

### PR DESCRIPTION
**Pull request by jenelyn0214**

### CODE REVIEW

**Comment by llamatester** [comment]
The wise-lending-v2 adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts fees wise-lending-v2
    
    🦙 Running WISE-LENDING-V2 adapter 🦙
    ---------------------------------------------------
    Start Date:	Sun, 06 Oct 2024 03:10:55 GMT
    End Date:	Mon, 07 Oct 2024 03:10:55 GMT
    ---------------------------------------------------
    
    ARBITRUM 👇
    Backfill start time: 1/10/2024
    NO METHODOLOGY SPECIFIED
    Daily fees: 352
    End timestamp: 1728270654 (2024-10-07T03:10:54.000Z)

**Review by Rezacb0413 — APPROVED** [review]

**Comment by Rezacb0413** [comment]
- [ ]

**Comment by vm06007** [comment]
@jenelyn0214 could you take the on-chain code we use for https://data.wisetoken.net/WISE/globals/fees.json and implement here same calls to contract data that gives us same result as in the feed.

**Comment by llamatester** [comment]
The wise-lending-v2 adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts fees wise-lending-v2
    
    🦙 Running WISE-LENDING-V2 adapter 🦙
    ---------------------------------------------------
    Start Date:	Wed, 09 Oct 2024 11:05:44 GMT
    End Date:	Thu, 10 Oct 2024 11:05:44 GMT
    ---------------------------------------------------
    
    ARBITRUM 👇
    Backfill start time: 1/10/2024
    NO METHODOLOGY SPECIFIED
    Daily fees: 346
    End timestamp: 1728558343 (2024-10-10T11:05:43.000Z)

**Comment by vm06007** [comment]
@g1nt0ki can you check again we've changed the implementation to be on-chain

File: fees/wise-lending-v2/index.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+];
+
+const fetch: any = async ({ api }: FetchOptions) => {
+    const feeAmounts = await api.multiCall({
+        abi: "function feeTokens(address) view returns (uint256)",
+        target: feeMaangerContract,
+        calls: arbitrumPools,
+    });
+
+    api.add(arbitrumPools, feeAmounts);
```
Comment (dtmkeng) [inline fees/wise-lending-v2/index.ts]: seem it total value not daily

**Review by dtmkeng — CHANGES_REQUESTED** [review]

File: fees/wise-lending-v2/index.ts (hunk 1 above)
Comment (vm06007) [reply fees/wise-lending-v2/index.ts]: if you query this endpoint daily then you get the values on daily basis

Comment (vm06007) [reply fees/wise-lending-v2/index.ts]: so this adapter shows our current fees generated by the protocol that are not withdrawn, as soon as protocol withdraws the fees this global value of fees accumulated is reduced accordingly. @dtmkeng feel free to open our contract and see how the fees are handled. 

- query feeTokens() to see accumulated fees for the pool that are not withdrawn, query feeTokens() at any time to get the amount of fees protocol holds at each point in time. 

- elaborate about daily? contract provides fees for each pool at every given moment every time you make a snapshot of this data (daily, monthly or each hour is up to you) we give you data on chain about fees we've collected at each point available from the protocol.

Comment (vm06007) [reply fees/wise-lending-v2/index.ts]: > seem it total value not daily

can you elaborate how it should be? did you check contract code? the fees reported are correct per pool, if they are withdrawn on daily basis call to this function would reflect that in value updated every time queried.

**Comment by vm06007** [comment]
@g1nt0ki could you take a look and progress this PR? We've changed the solution to be on-chain.

**Review by vm06007 — APPROVED** [review]

File: fees/wise-lending-v2/index.ts (hunk 1 above)
Comment (dtmkeng) [reply fees/wise-lending-v2/index.ts]: value you get from function it total value not daily

Comment (dtmkeng) [reply fees/wise-lending-v2/index.ts]: daily value from contract function should like this https://github.com/DefiLlama/dimension-adapters/blob/master/fees/crv-usd.ts

**Comment by llamatester** [comment]
The wise-lending-v2 adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts fees wise-lending-v2
    
    🦙 Running WISE-LENDING-V2 adapter 🦙
    ---------------------------------------------------
    Start Date:	Sun, 13 Oct 2024 21:03:02 GMT
    End Date:	Mon, 14 Oct 2024 21:03:02 GMT
    ---------------------------------------------------
    
    ARBITRUM 👇
    Backfill start time: 1/10/2024
    NO METHODOLOGY SPECIFIED
    Daily fees: 0
    Total fees: 360
    End timestamp: 1728939781 (2024-10-14T21:03:01.000Z)

**Comment by g1nt0ki** [comment]
@jenelyn0214 hmm, on second thought, this might show lower fees because if the protocol withdraws fees in the 24 hour period, it is subtracted right? is there some event log when the withdrawal happens so we can include it?

**Comment by vm06007** [comment]
> @jenelyn0214 hmm, on second thought, this might show lower fees because if the protocol withdraws fees in the 24 hour period, it is subtracted right? is there some event log when the withdrawal happens so we can include it?

yes let's include it if possible, we have event `ClaimedFeesWise()` 

```
        emit ClaimedFeesWise(
            underlyingTokenAddress,
            tokenAmount,
            block.timestamp
        );
```

emitted during claimWiseFees call.

**Comment by g1nt0ki** [comment]
okay, updated code the use that event: https://github.com/DefiLlama/dimension-adapters/commit/a2a5c6ab0c4f0c5162933ff9e671093ee861972f

**Pull request merged**